src/foam/nanos/fs/File.js

```diff
@@ -267,7 +282,11 @@ foam.CLASS({
       name: 'authorizeOnDelete',
       javaCode: `
         AuthService auth = (AuthService) x.get("auth");
-        if ( ! auth.check(x, "file.remove." + getId()) ) {
+        Boolean hasPermission = auth.check(x, "file.remove." + this.getId());
+        Subject subject = (Subject) x.get("subject");
+        User user = subject.getRealUser();
+
+        if ( ! hasPermission && user.getId() != this.getCreatedBy() ) {
```

Comment:
Please use subject.getUser() instead then check it against the file owner.
```suggestion
        User user = subject.getUser();

        if ( ! hasPermission && user.getId() != this.getOwner() ) {
```

Reply:
okay, thank you.